lib/src/doc/lives.rs

```diff
@@ -30,26 +36,66 @@ impl<'a> Document<'a> {
 			for lv in self.lv(opt, txn).await?.iter() {
 				// Create a new statement
 				let lq = Statement::from(lv);
-				// Check LIVE SELECT where condition
-				if let Some(cond) = lq.conds() {
-					// Check if this is a delete statement
-					let doc = match stm.is_delete() {
-						true => &self.initial,
-						false => &self.current,
-					};
-					// Check if the expression is truthy
-					if !cond.compute(ctx, opt, txn, Some(doc)).await?.is_truthy() {
-						continue;
-					}
+				// Get the event action
+				let met = if stm.is_delete() {
+					Value::from("DELETE")
+				} else if self.is_new() {
+					Value::from("CREATE")
+				} else {
+					Value::from("UPDATE")
+				};
+				// Check if this is a delete statement
+				let doc = match stm.is_delete() {
+					true => &self.initial,
+					false => &self.current,
+				};
+				// Ensure that a session exists on the LIVE query
+				let sess = lv.session.as_ref().ok_or(Error::UnknownAuth)?;
+				// Ensure that auth info exists on the LIVE query
+				let auth = lv.auth.clone().ok_or(Error::UnknownAuth)?;
+				// We need to create a new context which we will
+				// use for processing this LIVE query statement.
+				// This ensures that we are using the session
+				// of the user who created the LIVE query.
+				let mut lqctx = Context::background();
+				lqctx.add_value("auth", sess.pick(SD.as_ref()));
```

Comment:
Would be good to have these fields declared as statics somewhere

Reply:
Yeah this would be good, definitely!